app/contacts/contact_ftp.py

```diff
@@ -0,0 +1,214 @@
+        self.server = MyServer(user, self.contact_svc, self.file_svc, self.logger, self.host, self.port, self.user,
+                               self.pword, self.directory)
+
+    async def ftp_server_python_old(self):
+        await self.server.start(host=self.host, port=self.port)
+
+    async def ftp_server_python_new(self):
+        await self.server.run(host=self.host, port=self.port)
+
+
+class MyServer(aioftp.Server):
+    def __init__(self, user, contact, file, log, host_ip, port_in, username, password, user_dir,
+                 *,  max_con=256):
+        super().__init__(user, maximum_connections=max_con)
+        self.contact_svc = contact
+        self.file_svc = file
+        self.logger = log
+        self.host = host_ip
+        self.port = port_in
+        self.login = username
+        self.pword = password
+        self.directory = user_dir
+
+    @aioftp.ConnectionConditions(
+        aioftp.ConnectionConditions.login_required,
+        aioftp.ConnectionConditions.passive_server_started)
+    @aioftp.PathPermissions(aioftp.PathPermissions.writable)
+    async def stor(self, connection, rest, mode='wb'):
+
+        @aioftp.ConnectionConditions(
+            aioftp.ConnectionConditions.data_connection_made,
+            wait=True,
+            fail_code='425',
+            fail_info='Can not open data connection')
+        @aioftp.worker
+        async def stor_worker(self, connection, rest):
+            bytes_obj = b''
+            stream = connection.data_connection
+            del connection.data_connection
+            if connection.restart_offset:
+                file_mode = 'r+b'
+            else:
+                file_mode = mode
+            file_out = connection.path_io.open(real_path, mode=file_mode)
+
+            async with file_out, stream:
+                if connection.restart_offset:
+                    await file_out.seek(connection.restart_offset)
+                async for data in stream.iter_by_block(connection.block_size):
+                    bytes_obj += data
+
+            await self.handle_agent_file(split_path, bytes_obj, r_class)
+            connection.response('226', 'data transfer done')
+            del stream
+            return True
+
+        real_path, virtual_path = self.get_paths(connection, rest)
+        split_path = str(virtual_path).split('/')
+        self.logger.debug('File received: %s' % str(split_path[-1]))
+        r_class = CalderaServer(self.contact_svc, self.file_svc, self.logger, self.directory)
+
+        if await connection.path_io.is_dir(real_path.parent):
+            coro = stor_worker(self, connection, rest)
+            task = asyncio.create_task(coro)
+            connection.extra_workers.add(task)
+            code, info = '150', 'data transfer started'
+        else:
+            code, info = '550', 'path unreachable'
+        connection.response(code, info)
+        return True
+
+    async def handle_agent_file(self, file_name, file_bytes, r_class):
```

Comment:
the `file_name` variable here is really a list containing the split file path components, so we should rename it as such. Something like `split_file_path` maybe?